File: codeforces/1600~1699/1686/a.cc
#include <bits/stdc++.h>

using namespace std;

using i64 = long long;

void solve() {
  int n = 0;
  cin >> n;
  vector<int> a(n);
  for (int i = 0; i < n; ++i) cin >> a[i];

  int sum = accumulate(a.begin(), a.end(), 0);
  for (int i = 0; i < n; ++i) {
    int rest = sum - a[i];
    if (rest % (n-1) == 0 && rest / (n-1) == a[i]) {
      cout << "YES" << endl;
      return;
    }
  }

  cout << "NO" << endl;
}

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(0);

  int T = 0;
  cin >> T;
  for (int t = 0; t < T; ++t) {
    solve();
  }
  return 0;
}

File: codeforces/1600~1699/1686/a.rs
use std::io;
use std::str::FromStr;
use std::fmt::Debug;

fn read_int<T: FromStr>() -> T where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    io::stdin().read_line(&mut s).unwrap();
    s.trim().parse::<T>().expect("invalid input")
}

fn read_vec<T: FromStr>() -> Vec<T> where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    io::stdin().read_line(&mut s).unwrap();
    s.split(" ").map(|x| x.trim().parse().expect("invalid input")).collect()
}

fn solve() {
    let n = read_int::<usize>();
    let a = read_vec::<i32>();
    let sum: i32 = a.iter().sum();
    let m: i32 = (n-1) as i32;
    for i in 0..n {
        let rest = sum - a[i];
        if rest % m == 0 && rest / m == a[i] {
            println!("YES");
            return;
        }
    }
    println!("NO");
}

fn main() {
    let n = read_int::<i32>();
    for _t in 0..n {
        solve();
    }
}   
